torchvision/transforms/functional_tensor.py

```diff
@@ -914,7 +896,28 @@ def equalize(img: Tensor) -> Tensor:
 
     _assert_channels(img, [1, 3])
 
-    if img.ndim == 3:
-        return _equalize_single_image(img)
+    # img is a ...xCxHxW tensor
+    # temp is a ...xCxHxWx256 tensor
+    temp = img[..., None] == torch.arange(0, 256, dtype=torch.uint8, device=img.device)
+    height_dim, width_dim = -3, -2
+    hist = temp.sum(dim=[height_dim, width_dim])
+
+    step = (hist.sum(dim=-1) - hist[..., -1]) // 255
+    step = step[..., None]
+    if (step == 0).all():
+        return img
 
-    return torch.stack([_equalize_single_image(x) for x in img])
+    luts = (torch.cumsum(hist, dim=-1) + (step // 2)) // step
```

Comment:
I think there is a potential bug here. In the original implementation the `step` was a scalar so the if statement was enough to protect against division by zero. Here it's a C dimensional vector and if some of its values are zeroes it can lead to infs/nans.